### PR TITLE
New JSON test format

### LINKED ISSUE #25: Create new article lists with oldid for latest revision
The JSON files in articles contain lists of articles for each of a dozen languages. For the new server these should ideally
- also have the oldid of the latest revision of each of these titles
- also have the prefix in the JSON structure, so that we don't need to rely on the file name

An idea is to change the representation of each test to this:

``` javascript
[{prefix: "enwiki", title:"Foo", oldid:556366262}]
```

The oldid for each title can be retrieved with an API query like this:
http://en.wikipedia.org/w/api.php?action=query&prop=revisions&titles=Main%20Page&rvprop=ids&format=json

### PR DESCRIPTION
**Pull request by krishnan-chandra**
Should fix #25 for the most part - there are some things that don't work quite right as I said on the IRC - I'll commit in addition to this a list of URLs that fail for me, but I just wanted to make sure there's nothing wrong in my script.

### CODE REVIEW

File: articles/add_oldids.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    output_file_parts = article_list.split('.')
+    output_file_parts[0] += '-modified'
+    output_file_name = ".".join(output_file_parts)
+    with open(article_list) as f:
+        titles = json.loads(f.read())
+        # The language is the first capturing group in the regex
+        lang = file_regex.search(article_list).group(1)
+        for title in titles:
+            # url encode the title so that we don't get silly hits.
+            url = api_string.format(lang, quote(title.encode('utf-8')))
```
Comment (krishnan-chandra) [inline articles/add_oldids.py:28]: Not so sure if this is right...I think it is though.

Comment (gwicke) [inline articles/add_oldids.py:28]: At first sight this looks ok to me.

File: articles/add_oldids.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            try:
+                response = requests.get(url)
+                resp_json = response.json()
+                oldid = resp_json['query']['pages'].values()[0]['revisions'][0]['revid']
+                output_objects.append({'prefix': lang + 'wiki', 'title': title, 'oldid': oldid})
+                successes += 1
+            except KeyError:
+                print resp_json
+                failed_objects.append({'prefix': lang + 'wiki', 'title': title})
```
Comment (gwicke) [inline articles/add_oldids.py]: Can you also record the failed url here? Would make it easier to track down url encoding issues.

**Comment by gwicke** [comment]
Ping! I'd love to merge this, but first we should work out the issue with the failing API requests.

**Comment by krishnan-chandra** [comment]
My apologies for the delay - I have exams at the beginning of this week but I will have this resolved by the end of Wednesday.

**Comment by gwicke** [comment]
No worries, and good luck with the exams!

**Comment by krishnan-chandra** [comment]
0.4% of all URLs we have for testing failed. Is that significant enough for us to look into it?

**Comment by krishnan-chandra** [comment]
@gwicke I'm closing this, as I think it's complete. Please comment on my pull request if you have issues with it.

**Pull request merged**